src/ListGroupItem.js

```diff
@@ -33,6 +33,12 @@ const ListGroupItem = (props) => {
     'list-group-item'
   );
 
+  // Prevent click event when disabled.
+  if (disabled) {
```

Comment:
I was going to ask to convert this to a normal component to make use of instance methods like [onClick in Button](https://github.com/reactstrap/reactstrap/blob/master/src/Button.js#L32-L41) but this is basically the same thing. Neat! I believe a new function is assigned each time it is rendered as disabled. Can you move this logic to a `const handleDisabledOnClick` that's assigned outside of the `ListGroupItem` function.